main.nf

```diff
@@ -199,132 +258,304 @@ process get_software_versions {
+        if ((matcher = line =~ /Uniquely mapped reads %\s*\|\s*([\d\.]+)%/)) {
+            percent_aligned = matcher[0][1]
+        }
+    }
+    logname = logs.getBaseName() - 'Log.final'
+    if(percent_aligned.toFloat() <= '5'.toFloat() ){
+        log.info "#################### VERY POOR ALIGNMENT RATE! IGNORING FOR FURTHER DOWNSTREAM ANALYSIS! ($logname)    >> ${percent_aligned}% <<"
+        skipped_poor_alignment << logname
+        return false
+    } else {
+        log.info "          Passed alignment > star ($logname)   >> ${percent_aligned}% <<"
+        return true
+    }
+}
+process star {
+    label 'high_memory'
 
-  process run_alevin_qc {
     tag "$prefix"
-    publishDir "${params.outdir}/alevin_qc", mode: 'copy'
+    publishDir "${params.outdir}/STAR", mode: 'copy'
+
+    when:
+    params.aligner == "star"
 
     input:
-    file result from alevin_results
+    // TODO (Nurlan Kerimov):  change the prefix to samplename in the future (did not do it because there is no test environment for changes)
+    set val(samplename), file(reads) from read_files_star
+    file index from star_index.collect()
+    file gtf from gtf_star.collect()
+    file whitelist from barcode_whitelist.collect()
 
     output:
-    file "${name}_alevinqc_results" into alevinqc_results
+    set file("*Log.final.out"), file ('*.bam') into star_aligned
+    file "*.out" into alignment_logs
+    file "*SJ.out.tab"
+    file "*Log.out" into star_log
+    file "${prefix}Aligned.sortedByCoord.out.bam.bai" into bam_index_rseqc, bam_index_genebody
 
     script:
+    prefix = reads[0].toString() - ~/(_R1)?(_trimmed)?(_val_1)?(\.fq)?(\.fastq)?(\.gz)?$/
+    def star_mem = task.memory ?: params.star_memory ?: false
+    def avail_mem = star_mem ? "--limitBAMsortRAM ${star_mem.toBytes() - 100000000}" : ''
 
-    prefix = result.toString() - '_alevin_results'
-
+    seqCenter = params.seqCenter ? "--outSAMattrRGline ID:$prefix 'CN:$params.seqCenter'" : ''
+    cdna_read = reads[0]
+    barcode_read = reads[1]
     """
-    alevin_qc.r $result ${prefix} $result
+    STAR --genomeDir $index \\
+         --sjdbGTFfile $gtf \\
+         --readFilesIn $barcode_read $cdna_read  \\
+         --runThreadN ${task.cpus} \\
+         --twopassMode Basic \\
+         --outWigType bedGraph \\
+         --outSAMtype BAM SortedByCoordinate $avail_mem \\
+         --readFilesCommand zcat \\
+         --runDirPerm All_RWX \\
+         --outFileNamePrefix $prefix $seqCenter \\
+         --soloType Droplet \\
+         --soloCBwhitelist $whitelist
+
+    samtools index ${prefix}Aligned.sortedByCoord.out.bam
     """
 
-  }
+
+}
+// Filter removes all 'aligned' channels that fail the check
+star_aligned
+    .filter { logs, bams -> check_log(logs) }
+    .flatMap {  logs, bams -> bams }
+.into { bam_count; bam_rseqc; bam_preseq; bam_markduplicates; bam_htseqcount; bam_stringtieFPKM; bam_for_genebody; bam_dexseq; leafcutter_bam }
+
+
+
+ /*
+  * STEP 4 - Run alevin qc
```

Comment:
Yeah we can keep it that way for now - should integrate this later on when we have the final version of AlevinQC on bioconda...